### DISCUSSION TITLE
Make Button fill Grid Cell

### DISCUSSION THREAD

**Discussion by schneidermanuel** [AvaloniaUI/Avalonia#7331] [Q&A] [answered]
Hey everyone.

In WPF, I used to create a Button inside a Grid, and it sould automatically scale itself to fill the with of the Cell. 
In Avalonia, this is the case for controls, like a TextBox. For a Button, this does not work. 

This Xaml Code
```
<Grid Grid.Column="1" ShowGridLines="True">
<Grid.ColumnDefinitions >
		<ColumnDefinition />
		<ColumnDefinition />
		<ColumnDefinition />
	</Grid.ColumnDefinitions>
	<Grid.RowDefinitions>
		<RowDefinition />
		<RowDefinition />
		<RowDefinition />
	</Grid.RowDefinitions>
<Button Grid.Row="0" Grid.Column="0" />
</Grid>
```
results in this:
![image](https://user-images.githubusercontent.com/57318033/148696896-d9c7e015-2389-4b1b-afdf-4594fee75bb6.png)

while this code (just changed the button to a textBox)
```
<Grid Grid.Column="1" ShowGridLines="True">
<Grid.ColumnDefinitions >
		<ColumnDefinition />
		<ColumnDefinition />
		<ColumnDefinition />
	</Grid.ColumnDefinitions>
	<Grid.RowDefinitions>
		<RowDefinition />
		<RowDefinition />
		<RowDefinition />
	</Grid.RowDefinitions>
<TextBox Grid.Row="0" Grid.Column="0" />
</Grid>
```
results in this: 
![image](https://user-images.githubusercontent.com/57318033/148696911-f8745622-6608-4815-8837-a03c5d70d407.png)

The TextBox filles out all of the space, the button does not.

**Comment by maxkatz6** [comment, marked as answer]
HorizontalAlignment
VerticalAlignment 

Same as in WPF, different defaults.

**Reply by robloo** [reply]
In other words, set HorizontalAlignment and VerticalAlignment to Stretch on the Button.

**Comment by schneidermanuel** [comment]
oh ok, thanks. I never used those properties, I always used them as defaults. Thanks!
